.github/workflows/diff_debug.yaml

```diff
@@ -113,6 +113,13 @@ jobs:
           name: "Code coverage(Debug build)"
           path: tools/github/cppcheck_and_clang_format.txt
 
+      - name: Save drivers test report
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: "HA_drivers_report"
+          path: tests/drivers/test_report.tar.gz
+
```

Comment:
I would move this to be under the `Run HA driver tests` step, apart from that everything seems good

Reply:
But for other tests, uploading is also in different steps, I think it makes perfect sense that uploading is separated.

Reply:
Sorry for the confusion, I didn't express myself correctly. 
I agree it should be a separate step, however it should be moved to be right below (following the, after ...) the `Run HA driver tests` step. 
This is because artifacts get uploaded and can be viewed as soon as they are uploaded (actions/upload-artifact v4 version feature). So in a situation where someone wants to view the HA logs ASAP, they won't have to wait for `cppcheck and clang-format` step.